From 544fb39cb636ee9770e0ae9ddb3cce7a749cf529 Mon Sep 17 00:00:00 2001
From: Li Lingfeng <lilingfeng3@huawei.com>
Date: Mon, 6 Feb 2023 22:10:12 +0800
Subject: [PATCH 020/500] [Huawei] Revert "[Huawei] io_uring: fix soft lockup
 in io_submit_sqes()"

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188335

-------------------------------

This reverts commit af20e0feddf8ffc73666d039fe5e713a87d6d623.

We need to apply patch 788d0824269bef (io_uring: import 5.15-stable
io_uring) to move io_uring to separate directory and solve
the problem of CVE-2023-0240.
Revert this patch and add it again after patch 788d0824269bef to
reduce conflicts.

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
---
 fs/io_uring.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/fs/io_uring.c b/fs/io_uring.c
index 199de94fc3dc..d7391a1e0584 100644
--- a/fs/io_uring.c
+++ b/fs/io_uring.c
@@ -6706,9 +6706,6 @@ static int io_submit_sqes(struct io_ring_ctx *ctx, unsigned int nr)
 		err = io_submit_sqe(req, sqe, &link, &state.comp);
 		if (err)
 			goto fail_req;
-		/* to avoid doing too much in one submit round */
-		if (submitted > IORING_MAX_ENTRIES / 2)
-			cond_resched();
 	}
 
 	if (unlikely(submitted != nr)) {
-- 
2.17.1

